indev/libinput.c

```diff
@@ -419,30 +502,44 @@ static void read_pointer(libinput_drv_state_t *state, struct libinput_event *eve
   switch (type) {
     case LIBINPUT_EVENT_TOUCH_MOTION:
     case LIBINPUT_EVENT_TOUCH_DOWN:
+    case LIBINPUT_EVENT_TOUCH_UP:
       touch_event = libinput_event_get_touch_event(event);
+      /* ignore more than 2 fingers to avoid overflowing the buffer */
+      if (libinput_event_touch_get_slot(touch_event) > 1)
+        return;
+      evt = new_event(state);
```

Comment:
I think something has went missing here. `evt` is only set here for `LIBINPUT_EVENT_TOUCH_UP` but it's written into in all the other cases below?

Reply:
The other cases here fall through, I split this out as "common handling code", I'll rewrite it to be a bit clearer.

Reply:
oh I missed the LIBINPUT_EVENT_POINTER_* cases too... nice catch :sweat_smile: